nwb_conversion_tools/utils/common_writer_tools.py

```diff
@@ -0,0 +1,201 @@
+    ----------
+    dt: DynamicTable
+    prop_dict: dict
+        dict(name=dict(description='',data='', index=bool)
+    defaults: dict
+        default row values for dt columns
+
+    Returns
+    -------
+    default_updated: dict
+    """
+    defaults_updated = defaults
+    if dt is None:
+        return
+    property_default_data = DynamicTableSupportedDtypes
+    for colname in dt.colnames:
+        if colname not in defaults:
+            samp_data = dt[colname].data[0]
+            default_datatype = [proptype for proptype in property_default_data if isinstance(samp_data, proptype)][0]
+            defaults_updated.update({colname: property_default_data[default_datatype]})
+    # for all columns that are new for the given RX, they will
+    for name, des_dict in prop_dict.items():
+        des_args = dict(des_dict)
+        if name not in defaults_updated:
+            # build default junk values for data and add that as column directly later:
+            default_datatype_list = [
+                proptype for proptype in property_default_data if isinstance(des_dict["data"][0], proptype)
+            ][0]
+            des_args["data"] = [property_default_data[default_datatype_list]] * len(dt.id)
+            dt.add_column(name, **des_args)
+
+
+def check_module(nwbfile, name: str, description: str = None):
+    """
+    Check if processing module exists. If not, create it. Then return module.
+
+    Parameters
+    ----------
+    nwbfile: pynwb.NWBFile
+    name: str
+    description: str | None (optional)
+
+    Returns
+    -------
+    pynwb.module
+    """
+    assert isinstance(nwbfile, pynwb.NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
+    if name in nwbfile.processing:
+        return nwbfile.processing[name]
+    else:
+        if description is None:
+            description = name
+        return nwbfile.create_processing_module(name, description)
+
+
+def get_num_spikes(units_table, unit_id):
+    """Return the number of spikes for chosen unit."""
+    ids = np.array(units_table.id[:])
+    indexes = np.where(ids == unit_id)[0]
+    if not len(indexes):
+        raise ValueError(f"{unit_id} is an invalid unit_id. Valid ids: {ids}.")
+    index = indexes[0]
+    if index == 0:
+        return units_table["spike_times_index"].data[index]
+    else:
+        return units_table["spike_times_index"].data[index] - units_table["spike_times_index"].data[index - 1]
+
+
+def default_export_ops():
+    return dict(
+        use_times=False,
+        write_as="raw",
+        es_key=None,
+        buffer_gb=None,
+        buffer_shape=None,
+        chunk_mb=None,
+        iterator_type="v2",
+        chunk_shape=None,
+        write_scaled=False,
+        compression="gzip",
+        compression_opts=4,
+        skip_unit_properties=[],
+        skip_unit_features=[],
+        skip_electrode_properties=[],
+        unit_property_descriptions=dict(),
+        write_electrical_series=True,
+        overwrite=True,
+    )
```

Comment:
this should be a global dict. I don't see any reason why it should be a function.

```suggestion
default_export_ops =  dict(
    use_times=False,
    write_as="raw",
    es_key=None,
    buffer_gb=None,
    buffer_shape=None,
    chunk_mb=None,
    iterator_type="v2",
    chunk_shape=None,
    write_scaled=False,
    compression="gzip",
    compression_opts=4,
    skip_unit_properties=[],
    skip_unit_features=[],
    skip_electrode_properties=[],
    unit_property_descriptions=dict(),
    write_electrical_series=True,
    overwrite=True,
)
```

Reply:
Also, the null values should be omitted. I don't see why we should be passing empty lists and dictionaries.

Reply:
if its a global variable that I import in a script then if its updated, then there is no way to get the unaltered one again in that script right?

```python
from nwb_conversion_tools.utils.common_writer_tools import default_export_ops, default_export_ops_schema

conversion_ops = default_export_ops
 # will have to instead call: conversion_ops = dict(default_export_ops) each time
conversion_ops.update(kwargs)
```
I dont know, from a design point of view, is it better to call dict() each time to create a new instance of the global variable or have the global variable be the output of a function